frontend/packages/operator-lifecycle-manager/src/components/clusterserviceversion.tsx

```diff
@@ -517,19 +520,32 @@ export const NamespacedClusterServiceVersionList: React.SFC<ClusterServiceVersio
   );
   const NoDataEmptyMsg = () => <MsgBox title="No Operators Found" detail={noDataDetail} />;
 
-  const isCopiedCSV = (source, kind) => {
+  const isCopiedCSV = (source: ClusterServiceVersionKind | SubscriptionKind, kind: string) => {
     return (
       referenceForModel(ClusterServiceVersionModel) === kind && source.status?.reason === 'Copied'
     );
   };
 
-  const removeCopiedCSVs = (operators: (ClusterServiceVersionKind | SubscriptionKind)[]) => {
+  const isFailedNonStandardCSV = (operator: ClusterServiceVersionKind | SubscriptionKind) => {
+    return (
+      operator.metadata.annotations?.[operatorTypeAnnotation] === nonStandardAnnotationValue &&
+      operator.status?.phase === ClusterServiceVersionPhase.CSVPhaseFailed
+    );
+  };
+
+  const filterOperators = (
+    operators: (ClusterServiceVersionKind | SubscriptionKind)[],
+    allNamespaceActive: boolean,
+  ) => {
     return operators.filter((source) => {
       const kind = referenceFor(source);
-      if (isSubscription(kind)) {
+      if (isSubscription(source)) {
         return true;
       }
-      return !isCopiedCSV(source, kind);
+      if (allNamespaceActive) {
+        return !(isCopiedCSV(source, kind) || isFailedNonStandardCSV(source));
+      }
+      return !isFailedNonStandardCSV(source);
```

Comment:
Are only filtering out non-standard CSVs that are failed? The logic should be to filter non-standard CSVs that are *not* failed.